modules/imgcodecs/src/grfmt_tiff.cpp

```diff
@@ -804,6 +804,36 @@ bool  TiffEncoder::writeLibTiff( const Mat& img, const std::vector<int>& params)
         return false;
     }
 
+    //Get dpi.
+    int dpix = 300;
+    int dpiy = 300;
+
+    //Check for resolution inches.
+    int resUnit = RESUNIT_NONE;
+    readParam(params, IMWRITE_TIFF_RESUNIT_INCH, resUnit);
+
+    //Check for resolution centemeters.
+    if (resUnit == RESUNIT_NONE)
+    {
+        readParam(params, IMWRITE_TIFF_RESUNIT_CM, resUnit);
```

Comment:
I don't see any test, so I'm ask here: how units selection works?
Both are read into the same variable and passed to backed without any changes.